packages/synapse-react-client/src/components/SynapseChat/SynapseChat.tsx

```diff
@@ -53,51 +59,59 @@ export const SynapseChat: React.FunctionComponent<SynapseChatProps> = ({
+    },
   })
   // Restore chat session history, if exists.
   // TODO: currently only a single page is restored.  Add support for multiple pages (and detect the user scrolling up to restore the next page of results older)
-  const {
-    data: sessionHistoryInfiniteData,
-  } = //, hasNextPage: hasMoreSessionHistory, fetchNextPage: fetchNextSessionHistoryPage, isLoading: isSessionHistoryLoading } =
-    useGetAgentChatSessionHistoryInfinite(
-      {
-        sessionId: agentSession?.sessionId,
-      },
-      {
-        enabled: !!agentSession,
-      },
-    )
+  // const {
+  //   data: sessionHistoryInfiniteData,
+  // } = //, hasNextPage: hasMoreSessionHistory, fetchNextPage: fetchNextSessionHistoryPage, isLoading: isSessionHistoryLoading } =
+  //   useGetAgentChatSessionHistoryInfinite(
+  //     {
+  //       sessionId: agentSession?.sessionId,
+  //     },
+  //     {
+  //       enabled: !!agentSession,
+  //     },
+  //   )
 
-  const sessionHistory = useMemo(
-    () =>
-      sessionHistoryInfiniteData?.pages
-        .flatMap(page => page.page)
-        /* Note: session history is ordered from most recent to least recent in each page, so reverse the order when restoring the chat interface */
-        .reverse() ?? [],
-    [sessionHistoryInfiniteData],
-  )
+  // const sessionHistory = useMemo(
+  //   () =>
+  //     sessionHistoryInfiniteData?.pages
+  //       .flatMap(page => page.page)
+  //       /* Note: session history is ordered from most recent to least recent in each page, so reverse the order when restoring the chat interface */
+  //       .reverse() ?? [],
+  //   [sessionHistoryInfiniteData],
+  // )
 
   useEffect(() => {
-    // when we have both a message and response, add a new interaction and reset
+    // when we have both a message and either a response or response error, add a new interaction and reset
     // Note : We want the current interaction to be visible to the user while this async job processes (show the user message, and a loading response).
     // You might think we could do this work in the onSuccess of createAgentChatInteraction, but I think there's a race condition (currentInteraction may not be set when onSuccess is called).
-    if (currentResponse && pendingInteraction) {
+    if (pendingInteraction && (currentResponse || currentResponseError)) {
       pendingInteraction.chatResponseText = currentResponse
+      pendingInteraction.chatErrorReason = currentResponseError
       setInteractions([...interactions, pendingInteraction])
       setCurrentResponse('')
+      setCurrentResponseError('')
       setPendingInteraction(undefined)
     }
-  }, [currentResponse, pendingInteraction])
+  }, [currentResponse, currentResponseError, pendingInteraction])
 
   useEffect(() => {
     // on mount, create a new agent session!
-    if (createAgentSession && !agentSession) {
+    if (
+      createAgentSession &&
+      (!agentSession || agentSession.agentAccessLevel != agentAccessLevel)
```

Comment:
on agent access level change, create a new agent

Reply:
Probably add a confirmation dialog to tell the user that the context of their chat will be reset

Reply:
No need with https://rest-docs.synapse.org/rest/PUT/agent/session/sessionId.html

Reply:
That's the consequence that I brought up in the PLFM Issue Scoping Meeting that led to me finding out about https://rest-docs.synapse.org/rest/PUT/agent/session/sessionId.html